### PR TITLE
Fix build with -O3

### PR DESCRIPTION
**Pull request by LaszloLango**
Related issue: #657, #853 

JerryScript-DCO-1.0-Signed-off-by: László Langó llango.u-szeged@partner.samsung.com

### CODE REVIEW

**Comment by rtakacs** [comment]
Tested with gcc 5.2 and there are no errors.

**Comment by zherczeg** [comment]
I think we need to change the other side to clear the buffer in advance (with an uit32_t zero). The problem is, if normal buffer is passed, you may overwrite the next values in the buffer.

**Comment by LaszloLango** [comment]
@rtakacs, @zherczeg I've updated the PR. Please check.

File: jerry-core/ecma/builtin-objects/ecma-builtin-global.cpp [hunk 1]
```diff
@@ -1064,6 +1064,7 @@ ecma_builtin_global_object_encode_uri_helper (ecma_value_t uri, /**< uri argumen
   lit_code_point_t cp;
   ecma_char_t ch;
   lit_utf8_byte_t octets[LIT_UTF8_MAX_BYTES_IN_CODE_POINT];
+  memset (octets, LIT_BYTE_NULL , LIT_UTF8_MAX_BYTES_IN_CODE_POINT);
```
Comment (rtakacs) [inline jerry-core/ecma/builtin-objects/ecma-builtin-global.cpp]: Plus space after LIT_BYTE_NULL.

**Comment by rtakacs** [comment]
Checked, there are no build errors.

**Comment by LaszloLango** [comment]
@rtakacs, thanks. @zherczeg, please check.

**Comment by zherczeg** [comment]
I like this much better. LGTM

**Pull request merged**